deeprankcore/trainer.py

```diff
@@ -23,11 +23,11 @@ class Trainer():
     def __init__( # pylint: disable=too-many-arguments # noqa: MC0001
                 self,
                 neuralnet = None,
-                dataset_train: Union[GraphDataset, GridDataset] = None,
-                dataset_val: Union[GraphDataset, GridDataset] = None,
-                dataset_test: Union[GraphDataset, GridDataset] = None,
-                val_size: Union[float, int] = None,
-                test_size: Union[float, int] = None,
+                dataset_train: Union[GraphDataset, GridDataset,None] = None,
```

Comment:
It actually boils down to the same thing, but I think in most (or all) of our cases, the way Giulia suggested it is preferred

See the comment to [this post](https://stackoverflow.com/questions/69185886/preferred-python-typing-syntax-of-optionalunion-vs-unionnone), stating:

> I think it depends on the context, but I generally prefer Optional[...] in most cases where None is a default value for an argument that may or may not be provided a non-trivial value. But there are also cases where an argument or variable might not be "optional" where None is a valid value it can take. 

I noticed that python 10 has a [nicer way of dealing with Union](https://peps.python.org/pep-0604/), but I am currently running python 3.9 and I think that we don't want to force 3.10 on users, right?

Reply:
Exactly I've already seen it and it is nicer, but since we're still supporting python 3.9 we must keep using Optional.